### PR TITLE
NO-ISSUE: Log informative error in case subscription doesn't have CSV

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: src/k8s_client/k8s_client.go [hunk 1]
```diff
@@ -598,12 +598,16 @@ func (c *k8sClient) CreateEvent(namespace, name, message, component string) (*v1
...
 func (c *k8sClient) GetCSVFromSubscription(namespace string, name string) (string, error) {
-	result, err := c.olmClient.Subscriptions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	subscription, err := c.olmClient.Subscriptions(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
 		c.log.WithError(err).Warnf("Failed to get subscription %s in %s", name, namespace)
 		return "", err
 	}
-	return result.Status.CurrentCSV, nil
+	if subscription.Status.CurrentCSV == "" {
+		c.log.WithError(err).Warnf("subscription CurrentCSV is empty, subscription status: %+v", subscription.Status)
```
Comment (tsorya) [inline src/k8s_client/k8s_client.go:607]: return error, it should not happen i assume ?

Comment (omertuc) [reply src/k8s_client/k8s_client.go:607]: It does happen, while we wait

Comment (tsorya) [reply src/k8s_client/k8s_client.go:607]: so not better to return error in that case?

Comment (omertuc) [reply src/k8s_client/k8s_client.go:607]: hmm not sure. Depends on how the users of this function behave

Comment (eranco74) [reply src/k8s_client/k8s_client.go:607]: It get called in operatorHandler `IsInitialized` and `GetStatus` in both cases it's eventually wrapped in a loop.
I don't want to change behavior here. just add logging so it's clear that the CSV is empty and whats the subscription status

**Comment by eranco74** [comment]
devscript setup failed
/test edge-e2e-ai-operator-ztp

File: src/k8s_client/k8s_client.go [hunk 2]
```diff
@@ -598,12 +598,17 @@ func (c *k8sClient) CreateEvent(namespace, name, message, component string) (*v1
...
```
Comment (rccrdpccl) [inline src/k8s_client/k8s_client.go:607]: variable `err` passed to `WithError` will always be nil. We could reduce noise by not passing error context

**Comment by eranco74** [comment]
packet issue
/retest-required

**Comment by eranco74** [comment]
devscripts-setup failed
/retest-required

**Pull request merged**